cmd/forwarder/http.go

```diff
@@ -147,6 +150,18 @@ func (s *httpServer) Run() error {
 			defer body.Close()
 		}
 
+		buf := new(bytes.Buffer)
+		// This should only be reached if authentication information is valid.
+		authUser, _, _ := r.BasicAuth()
```

Comment:
Suggest doing something like this:

```go
u, _, ok := r.BasicAuth()
if ok && s.Config.LogAuthUser(u) {
    ....
}
```

Then change `LogAuthUser` to do all the checks like so:

```go
func (c *IssConfig) LogAuthUser(u string) bool {
    return c.ValidTokenUser != "" &&
        u != c.ValidTokenUser &&
        rand.Intn(99)+1 >= c.TokenUserSamplePct
}
```

Reply:
Oh I dig that.